spring-integration-core/src/main/java/org/springframework/integration/channel/management/ChannelSendMetrics.java

```diff
@@ -79,33 +83,38 @@ public String getName() {
 		return name;
 	}
 
-	public StopWatch beforeSend() {
+	public long beforeSend() {
 		if (logger.isTraceEnabled()) {
 			logger.trace("Recording send on channel(" + this.name + ")");
 		}
-		final StopWatch timer = new StopWatch(this.name + ".send:execution");
-		timer.start();
 
+		long start = 0;
+		if (this.fullStatsEnabled) {
+			start = System.currentTimeMillis();
+			this.sendRate.increment();
+		}
 		this.sendCount.incrementAndGet();
-		this.sendRate.increment();
 
-		return timer;
+		return start;
 	}
 
-	public void afterSend(StopWatch timer, boolean result) {
-		if (timer != null) {
-			timer.stop();
-			if (result) {
+	public void afterSend(long start, boolean result) {
+		if (start > 0) {
+			long now = System.currentTimeMillis();
```

Comment:
Looks like this `now` should be withing `if (result && this.fullStatsEnabled)` as well.